### ISSUE TITLE
An error occurred while Style/ArgumentsForwarding cop was inspecting

### ISSUE THREAD

**Issue by tim-semba** [rubocop/rubocop#12832]
## Expected behavior

No errors occur.

## Actual behavior

```
1 error occurred:
An error occurred while Style/ArgumentsForwarding cop was inspecting foobar.rb:6:4.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.62.1 (using Parser 3.3.0.5, rubocop-ast 1.31.2, running on ruby 3.3.0) [x86_64-linux]
```

## Steps to reproduce the problem

1. Put foobar.rb

```ruby
# frozen_string_literal: true

# Foobar class
class Foobar
  class << self
    def foobar(foo)
      class << foo
        def bar(baz, &)
          baz.some_method(&)
        end
      end
    end
  end
end
```

2. Put .rubocop.yml

```
AllCops:
  TargetRubyVersion: 3.3
  NewCops: enable
```

3. Run

```
$ bundle exec rubocop -c .rubocop.yml foobar.rb
```

## RuboCop version

```
$ bundle exec rubocop -V
1.62.1 (using Parser 3.3.0.5, rubocop-ast 1.31.2, running on ruby 3.3.0) [x86_64-linux]
  - rubocop-performance 1.21.0
  - rubocop-rails 2.24.1
  - rubocop-rspec 2.29.1
```
